graphs/eacs/src/test/kotlin/com/anaplan/engineering/azuki/graphs/eacs/BEH7.kt

```diff
@@ -0,0 +1,136 @@
+
+@BEH(GraphBehaviours.GetSimpleCycleCount, GraphFunctions.GetSimpleCycleCount, """
+    Determine the number of cycles in a given directed graph
+""")
+class BEH7 : GraphScenario() {
+
+    @Eac("A cycle is a continuous path which starts and ends at the same vertex")
+    fun hasOneCycleDirected() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("d", "a")
+                edge("e", "d")
+                edge("c", "e")
+            }
+        }
+        then {
+            getSimpleCycleCount(graphA, 1)
+        }
+    }
+
+    @Eac("When there is no continuous path which starts and ends at the same vertex there are no cycles")
+    fun hasNoCyclesDirected() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("d", "a")
+                edge("e", "d")
+            }
+        }
+        then {
+            getSimpleCycleCount(graphA, 0)
+        }
+    }
+
+    @Eac("""
+        When the given graph is directed, any path which would be considered a cycle in an undirected graph but
+        where two edges start (or end) at the same vertex is not counted as a cycle
+        """)
+    fun hasReversedEdgeDirected() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("d", "a")
+                edge("e", "d")
+                edge("e", "c")
+            }
+        }
+        then {
+            getSimpleCycleCount(graphA, 0)
+        }
+    }
+
+    @Eac("When the given graph is empty, return 0")
+    fun hasEmptyDirectedGraph() {
+        given {
+            thereIsADirectedGraph(graphA) {}
+        }
+        then {
+            getSimpleCycleCount(graphA, 0)
+        }
+    }
+
+    @Eac("When the given graph has multiple individual cycles that share edges, each cycle is counted")
+    fun hasSubCircleDirected() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("d", "a")
+                edge("e", "d")
+                edge("c", "f")
+                edge("f", "e")
+                edge("b", "e")
+            }
+        }
+        then {
+            getSimpleCycleCount(graphA, 2)
+        }
+    }
+
+    @Eac("When there is a continuous path that only goes to one other vertex in the given graph, a cycle is counted")
+    fun hasEdgesInBothDirections() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "a")
+            }
+        }
+        then {
+            getSimpleCycleCount(graphA, 1)
+        }
+    }
+
+    @Eac("""
+        When there is a continuous path that goes through multiple vertices it will only visit each vertex once
+        """)
```

Comment:
🚫 This is a trivial consequence of the fact that the function counts simple cycles, so this isn't really a separate AC. I would however recommend using this scenario for that EAC instead of `hasOneCycleDirected`.